tests/uanets/models/test_mc_dropout.py

```diff
@@ -142,3 +148,65 @@ def test_dropout_network_accepts_scalars() -> None:
     test_points = tf.linspace(-1, 1, 100)
 
     assert model(test_points).shape == (100, 1)
+
+
+@pytest.mark.parametrize(
+    "input_shape, output_shape",
+    [
+        [[1, 1], [1, 1]],
+        [[10, 5], [10, 3]],
+    ],
+)
+def test_dropout_network_predict_call_shape(
+    input_shape: List[int], output_shape: List[int], rate: float
+) -> None:
+    inputs, outputs = random_inputs_outputs(input_shape, output_shape)
+
+    model = mc_dropout_model(inputs, outputs, rate)
+    predicted_means, predicted_vars = model.predict_mean_and_var(inputs, 100)
+
+    assert tf.is_tensor(predicted_means)
+    assert predicted_means.shape == output_shape
+    assert tf.is_tensor(predicted_vars)
+    assert predicted_vars.shape == output_shape
+
+
+@pytest.mark.parametrize("num_samples", [6, 12])
+@pytest.mark.parametrize(
+    "input_shape, output_shape",
+    [
+        [[1, 1], [1, 1]],
+        [[10, 5], [10, 3]],
+    ],
+)
+def test_dropout_network_sample_call_shape(
+    input_shape: List[int], output_shape: List[int], num_samples: int, rate: float
+) -> None:
+    inputs, outputs = random_inputs_outputs(input_shape, output_shape)
+
+    model = mc_dropout_model(inputs, outputs, rate)
+
+    samples = model.sample(inputs, num_samples)
+
+    assert tf.is_tensor(samples)
+    assert samples.shape == [num_samples] + output_shape
+
+
+@random_seed
+@pytest.mark.parametrize("num_samples", [100, 500, 1000])
+def test_dropout_network_sample_equal_predict(rate: float, num_samples) -> None:
+    input_shape, output_shape = [100, 5], [100, 3]
+    inputs, outputs = random_inputs_outputs(input_shape, output_shape)
+
+    model = mc_dropout_model(inputs, outputs, rate)
+
+    samples = model.sample(inputs, num_samples)
+    sample_mean = tf.reduce_mean(samples, axis=0)
+    sample_variance = tf.reduce_mean((samples - sample_mean) ** 2, axis=0)
```

Comment:
```suggestion
    sample_variance = tf.math.reduce_variance(samples, axis=0)
```
Actually, could we just use `reduce_variance` in the actual `predict_mean_and_var` implementation?

Reply:
that looks correct